pkg/kubelet/kubelet.go

```diff
@@ -1701,6 +1709,50 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	return nil
 }
 
+func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) {
+	podResized := false
+	for _, container := range pod.Spec.Containers {
+		if len(diff.ObjectDiff(container.ResourcesAllocated, container.Resources.Requests)) > 0 {
+			podResized = true
+			break
+		}
+	}
+	if !podResized {
+		return
+	}
+
+	var otherActivePods []*v1.Pod
+	activePods := kl.GetActivePods()
+	for _, p := range activePods {
+		if p.UID != pod.UID {
+			otherActivePods = append(otherActivePods, p)
+		}
+	}
+	if ok, failReason, failMessage := kl.canAdmitPod(otherActivePods, pod); !ok {
```

Comment:
is this function thread safe, and also, will this ActivePods, which was retrieved from the runtimeCache, which can be slight different with the exact current PODs ( with incomings from kubelet ). how do you plan to handle situations as such ?

Reply:
very good point.. this may be a problem. let me investigate this more. thanks for catching this!

Reply:
this was indeed an issue. i've modified it to serialize with a mutex and i'm testing it now, will send an updated change out tomorrow.

Reply:
excellent!

Reply:
i pushed a commit just now that fixes this for add vs. resize race, as well as multiple resize races.